### ISSUE TITLE
Question regarding nb_epoch

### ISSUE THREAD

**Issue by cbaziotis** [maxpumperla/hyperas#63]
Hello.
I have a question about `nb_epoch`. 
In order to get a more accurate estimation about the performance of each model i want to let each one train for many epochs, lets say 10. How does hyperas calculate the performance of each model? 

1. Based on the best epoch or the last epoch?
2. What metric does it use (training loss, validation loss...)?

**Comment by pianomanu** [comment]
I'd say the validation loss of the last epoch

**Comment by BenPorebski** [comment]
Hyperas calculates the performance of each model based on whatever you return as the 'loss'.

I personally use a larger nb_epoch and a keras checkpoint and early stopper callback to temporarily save the best performing model.

I then load that model, calculate a metric (in this case the R^2 of the fit) and return that as the 'loss'.

```
    checkpointer = ModelCheckpoint(filepath="1dcnn_best.h5", verbose=1, save_best_only=True, save_weights_only=True)
    earlystopper = EarlyStopping(monitor="val_loss", patience=5, verbose=1)

    model.fit(X_train, y_train, batch_size=128, nb_epoch=50, shuffle=True, validation_data=(X_test, y_test), callbacks=[checkpointer, earlystopper])
    model.load_weights("1dcnn_best.h5")

    pred = model.predict(X_test)
    score = r2_score(y_test, pred)
    return {'loss': -score, 'status': STATUS_OK, 'model': model}

```

**Comment by maxpumperla** [comment]
this has been sufficiently answered by @BenPorebski. Whatever metric you're using is dependent on your problem domain, I can't answer in this generality.

**Comment by gledsonmelotti** [comment]
Hello, how are you? I have some questions about the commands checkpointer = ModelCheckpoint (filepath = "1dcnn_best.h5", verbose = 1, save_best_only = True, save_weights_only = True) and earlystopper = EarlyStopping (monitor = "val_loss", patience = 5, verbose = 1) .

For example, if I use save_best_only = True, save_weights_only = True in ModelCheckpoint, am I saving only the best set of weights or am I saving the best model with the best set of weights?

If I use at the end of the training (even using the ModelCheckpoint with save_best_only = True and save_weights_only = True) model.save ('Pedestre_PedestreNao_model.h5') and model.save_weights ('Pedestre_PedestreNao_weights.h5') I'm getting the best model and the best set of weights or am I saving the last model and the last set of weights (regardless if it is the best or not model and weights)?

I ask this question because I need to save the best model (one file) and the set of best weights (another file).

On EarlyStopping, does it stop the training for one epoch and move on to another epoch or stop the entire training (regardless of which epoch the model is)?

Thanks.
